pages/blog/introducing-envelop.mdx

```diff
@@ -0,0 +1,310 @@
+      res.send({ errors: [{ message: 'Not Supported in this demo' }] });
+    }
+  },
+});
+
+app.listen(3000, () => {
+  console.log(`GraphQL server is running...`);
+});
+```
+
+With that example, we only used `useLogger`, so while executing GraphQL operations, you should see that everything you do should be printed to the log.
+
+## Use Plugins
+
+![Plugins](https://www.envelop.dev/assets/features-pluggable.png)
+
+But logging is not everything possible with Envelop. By adding more plugins, you can add more features to your GraphQL execution, based on your app needs.
+
+For example, here’s a cool snippet for boosting things in your execution layer:
+
+```ts
+const getEnveloped = envelop({
+  plugins: [
+    useSchema(schema),
+    useParserCache(),
+    useValidationCache(),
+    useGraphQLJit(),
+  ],
+});
+```
+
+> By using `useParserCache` we make sure to parse every unique operation only once.
+> By using `useValidationCache` we make sure to validate every unique operation only once.
+> By using `useGraphQLJit` we replace the default execute function with a [just-in-time implementation](https://github.com/zalando-incubator/graphql-jit).
+
+While working without clients, we saw that many pieces of code can be moved into an Envelop plugin, and shared with the community. That created tons of plugins that you can now use quickly, without implementing it on your own for that specific project!
+
+We also created [Envelop Plugins Hub](https://www.envelop.dev/plugins) aa place where you can find all the plugins that are available for Envelop, with their documentation, versions, and some stats. Plugin Hub is open and available for the community to add their own.
+
+## Write your own plugins
+
+![Plugins](https://www.envelop.dev/assets/features-modern.png)
+
+Writing plugins for Envelop is super simple. We allow you to write code that connects to the phases that you need, and we’ll make sure to run your functions at the right time.
+
+Plugins can either live as internal plugins that are relevent only to your project, or you can share it with the community as NPM package.
+
+To get started with a custom plugin, choose what phases you need, and create functions that handle what you need. Envelop will provide a low-level, flexible api in each phase, so you can communicate with the core pipeline.
+
+```ts
+import { Plugin } from '@envelop/types';
+
+const myPlugin: Plugin = {
+  onParse({ params }) {
+    console.log('Parse started!', { args });
+
+    return (result) => {
+      console.log('Parse done!', { result });
+    };
+  },
+  onExecute({ args }) {
+    console.log('Execution started!', { args });
+
+    return {
+      onExecuteDone: ({ result }) => {
+        console.log('Execution done!', { result });
+      },
+    };
+  },
+};
+
+const getEnveloped = envelop({
+  plugins: [
+    /// ... other plugins ...,
+    myPlugin,
+  ],
+});
+```
+
+**[You can find here the complete plugins documentation](https://www.envelop.dev/docs/plugins)**
+
```

Comment:
What other ideas do you think you would come up with?

Here is an example of another cool plugin coming soon - [Operation Migration plugin](https://github.com/dotansimha/envelop/pull/94) - prevent GraphQL breaking changes by creating GraphQL Migrations!!

Reply:
or another one - [public and private schema filter](https://github.com/n1ru4l/graphql-public-schema-filter) - filter your GraphQL schema operations using directives